Add documentation about Khepri (#2058)
rabbitmq/rabbitmq-website: docs/metadata-store/clustering.md

same way as with Mnesia.

<figure className={diagramStyles.diagram}>
<MetadataStoreCluster/>
<figcaption>Clustering RabbitMQ happens at the metadata store level</figcaption>
</figure>

## Creating a cluster

You can create a cluster using the regular methods:
* using the CLI
* using peer discovery

See the [clustering guide](../clustering) for a complete description.

Here is an example using the CLI:

```
rabbitmqctl join_cluster rabbit@remote-host
```

## Removing a node from a cluster

Again, the regular methods can be used to remove a node from a cluster. Here
is an example with the CLI:

```
rabbitmqctl stop_app
rabbitmqctl reset
rabbitmqctl start_app
```

## When to enable Khepri?

It is possible to enable Khepri before or after you create or expand the
RabbitMQ. When Khepri is enabled, the first step is to [synchronize the
cluster view from Mnesia to
Khepri](./enabling-khepri#cluster-membership-synchronization) if Khepri was
enabled before the cluster was created.

Also, if you want to cluster nodes where some of them are using Mnesia and
some Khepri, the effective backend in the end will depend on several

> conditions, like if some nodes are already clustered. You will have to verify

Reviewer: This paragraph about clustering a mix of khepri-enabled and not enabled nodes is confusing to me. Do we need to discuss this at all? If so, should we move this to a dedicated paragraph and elaborate when would someone need/want to do this?

Author: Ok, I deleted it for now.

ater the change to clustering which backend was selected.

## Caveats

> Because of the use of the Raft consensus algorithm, you need to keep in mind

Reviewer: ```suggestion
Because of the use of the Raft consensus algorithm,
```

> that all operations that involve an update to the metadata store — and

Reviewer: ```suggestion
all operations that involve an update to the metadata store — and
```

sometimes even queries — require that a quorum number of nodes are available.

<figure className={diagramStyles.diagram}>
<MetadataStoreMinority/>
<figcaption>A quorum is required for store updates and cluster changes</figcaption>
</figure>

### Restarting a cluster member

When a cluster member is restarted or stopped, the remaining nodes may not
form a quorum anymore. This may affect the ability to start a node.

> For example, in a cluster of e.g. 5 nodes where all nodes are stopped, the

Reviewer: ```suggestion
For example, in a cluster of nodes where all nodes are stopped, the
```

> first two starting nodes will wait for a third node to start before completing

Reviewer: ```suggestion
first two starting nodes will wait for the third node to start before completing
```

> their boot and start serving messages. That’s because the metadata store needs

Reviewer: ```suggestion
their boot and starting to serve messages. That’s because the metadata store needs
```

> at least 3 node in this example to elect a leader and finich to initialize its

Reviewer: ```suggestion
at least 3 nodes in this example to elect a leader and finich to initialize its
```

> state to a known good one. In the meantime the first two nodes wait and may

Reviewer: ```suggestion
In the meantime the first two nodes wait and may
```
